src/System.Private.ServiceModel/src/System/ServiceModel/ActionNotSupportedException.cs

```diff
@@ -4,15 +4,18 @@
 
 
 using System.Diagnostics.Contracts;
+using System.Runtime.Serialization;
 using System.ServiceModel.Channels;
 
 namespace System.ServiceModel
 {
+    [Serializable]
     public class ActionNotSupportedException : CommunicationException
     {
         public ActionNotSupportedException() { }
         public ActionNotSupportedException(string message) : base(message) { }
         public ActionNotSupportedException(string message, Exception innerException) : base(message, innerException) { }
+        protected ActionNotSupportedException(SerializationInfo info, StreamingContext context) : base(info, context) { }
```

Comment:
I am wondering if we need to add this ctor to the contract as `ActionNotSupportedException` is in S.SM.Primitives contract. @mconnew, what do you think?